Mage.Sets/src/mage/sets/fallenempires/DelifsCube.java

```diff
@@ -0,0 +1,119 @@
+ *
+ *  The views and conclusions contained in the software and documentation are those of the
+ *  authors and should not be interpreted as representing official policies, either expressed
+ *  or implied, of BetaSteward_at_googlemail.com.
+ */
+package mage.sets.fallenempires;
+
+import java.util.UUID;
+import mage.abilities.Ability;
+import mage.abilities.common.AttacksAndIsNotBlockedTriggeredAbility;
+import mage.abilities.common.SimpleActivatedAbility;
+import mage.abilities.costs.common.RemoveCountersSourceCost;
+import mage.abilities.costs.common.TapSourceCost;
+import mage.abilities.costs.mana.ManaCostsImpl;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.RegenerateTargetEffect;
+import mage.abilities.effects.common.continuous.AssignNoCombatDamageSourceEffect;
+import mage.abilities.effects.common.continuous.GainAbilityTargetEffect;
+import mage.cards.CardImpl;
+import mage.constants.CardType;
+import mage.constants.Duration;
+import mage.constants.Outcome;
+import mage.constants.Rarity;
+import mage.constants.Zone;
+import mage.counters.CounterType;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.target.common.TargetControlledCreaturePermanent;
+
+/**
+ *
+ * @author MarcoMarin
+ */
+public class DelifsCube extends CardImpl {
+
+    public DelifsCube(UUID ownerId) {
+        super(ownerId, 170, "Delif's Cube", Rarity.RARE, new CardType[]{CardType.ARTIFACT}, "{1}");
+        this.expansionSetCode = "FEM";
+
+        Ability ability2 = new AttacksAndIsNotBlockedTriggeredAbility(new DelifsCubeEffect(this.getId()), false);        
+        ability2.addEffect(new AssignNoCombatDamageSourceEffect(Duration.EndOfTurn));        
+        // {2}, {tap}: This turn, when target creature you control attacks and isn't blocked, it assigns no combat damage this turn and you put a cube counter on Delif's Cube.
+        SimpleActivatedAbility ability = new SimpleActivatedAbility(Zone.BATTLEFIELD,
+                new GainAbilityTargetEffect(ability2, Duration.EndOfTurn),
+                new ManaCostsImpl("{2}"));
+        ability.addCost(new TapSourceCost());
+        ability.addTarget(new TargetControlledCreaturePermanent());
+        
+        this.addAbility(ability);
+        // {2}, Remove a cube counter from Delif's Cube: Regenerate target creature.
+        SimpleActivatedAbility ability3 = new SimpleActivatedAbility(Zone.BATTLEFIELD,
+                new RegenerateTargetEffect(),
+                new ManaCostsImpl("{2}"));
+        ability3.addCost(new RemoveCountersSourceCost(CounterType.CUBE.createInstance()));
+        ability3.addTarget(new TargetControlledCreaturePermanent());
+        
+        this.addAbility(ability3);
+    }
+
+    public DelifsCube(final DelifsCube card) {
+        super(card);
+    }
+
+    @Override
+    public DelifsCube copy() {
+        return new DelifsCube(this);
+    }
+}
+
+class DelifsCubeEffect extends OneShotEffect{
```

Comment:
I can't because of that same reason above.

 I know it's kinda indirect/strange, but I think it's the best way to keep with the modular philosophy of Mage. (cook as much as possible there in the constructor with already made parts) If I use source, the new source will be the targeted creature (who is gaining that ability).

I tested most cards this time. :) This one works. (I marked the tested ones in the Card tracking site)